ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -775,17 +809,14 @@ impl<'a> Assemble<'a> {
                         // The value is a integer and thus can't be stored in a float register.
                         panic!()
                     }
-                    VarLocation::Direct { frame_off, size } => {
+                    VarLocation::Direct { frame_off, size: _ } => {
                         // FIXME If we prime the register allocator with the interpreter frames
                         // stack size and don't create a new frame in the trace epilogue we can
                         // reference values directly using the current RBP value.
-                        dynasm!(self.asm; mov Rq(tgt_reg.code()), QWORD [Rq(Rq::RBP.code())]);
-                        match size {
-                            8 => {
-                                dynasm!(self.asm; lea Rq(tgt_reg.code()), [Rq(tgt_reg.code()) + frame_off])
-                            }
-                            _ => todo!(),
-                        }
+                        dynasm!(self.asm
```

Comment:
How comes we no longer need to inspect the size here?

Reply:
I don't understand this code. It seems to work, but I don't really understand why the `lea` exists.

Reply:
Oh, now I get it! These trace inputs become `ptr`s e.g. `%0: ptr = load_ti Direct(6, -32, 8)`.

Reply:
So I misunderstood how this worked earlier, but now I get it. Restored/fixed in 6f7157c28bfd46bfa4de6927a8c78b6675af0fe8.

Reply:
Right. `lea` is "load effective address". From memory it was designed to do certain pointer arithmetic operations to be done in one instruction (without explicit `add` and `mul`).

Nowadays people abuse it to do all kind of things :P